examples/src/main/java/com/datastax/oss/driver/examples/mapper/killrvideo/user/CreateUserQueryProvider.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.examples.mapper.killrvideo.user;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.mapper.MapperContext;
+import com.datastax.oss.driver.api.mapper.entity.EntityHelper;
+import com.datastax.oss.driver.api.mapper.entity.saving.NullSavingStrategy;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Provides the implementation of {@link UserDao#create}.
+ *
+ * <p>Package-private visibility is sufficient, this will be called only from the generated DAO
+ * implementation.
+ */
+class CreateUserQueryProvider {
+
+  private final CqlSession session;
+  private final EntityHelper<User> userHelper;
+  private final EntityHelper<UserCredentials> credentialsHelper;
+  private final PreparedStatement preparedInsertCredentials;
+  private final PreparedStatement preparedDeleteCredentials;
+  private final PreparedStatement preparedInsertUser;
+
+  CreateUserQueryProvider(
+      MapperContext context,
+      EntityHelper<User> userHelper,
+      EntityHelper<UserCredentials> credentialsHelper) {
+
+    this.session = context.getSession();
+
+    this.userHelper = userHelper;
+    this.credentialsHelper = credentialsHelper;
+
+    this.preparedInsertCredentials =
+        session.prepare(credentialsHelper.insert().ifNotExists().asCql());
+    this.preparedDeleteCredentials =
+        session.prepare(credentialsHelper.deleteByPrimaryKey().asCql());
+    this.preparedInsertUser = session.prepare(userHelper.insert().ifNotExists().asCql());
+  }
+
+  Optional<User> create(String firstname, String lastname, String email, char[] password) {
```

Comment:
The fact that Optional<User>.empty() is returned when the actual user is present in the db seems a bit odd.
I wonder if there are other options or this is an idiomatic way to handle that?

Reply:
The reason I wanted to return the user is because the method generates the id. But since creating an id is trivial anyway, maybe we can take it as a parameter, and the method will just return a boolean.